### ISSUE TITLE
`g:lsp_ale_auto_enable_linter` ignored?

### ISSUE THREAD

**Issue by dubst3pp4** [rhysd/vim-lsp-ale#6]
Hello,
I've configured ALE to use `vls` to lint `.vue` files, so in `after/ftplugins/vue.vim` I've set:

```vim
" use vls instead of tsserver to ensure correct linting of typescript in vue
" files, see https://github.com/dense-analysis/ale/issues/927#issuecomment-829533341
let b:ale_linters = ['eslint', 'vls']
let b:ale_fixers = ['eslint', 'vls']
```

Because of `vls` I want to disable automatic enablement of `vim-lsp` in ALE. So I put the following into my `.vimrc`:

```vim
let g:lsp_ale_auto_enable_linter = 0
```

But when I now open a `.vue` file and type in `:ALEInfo`, ALE now tells me that `vls` is  disabled:

```
 ALEInfo
 Current Filetype: vue
Available Linters: ['cspell', 'vls' ...]
  Enabled Linters: ['vls', 'eslint']
  Ignored Linters: ['vls']
```

Linting with `vls` doesn't work anymore. It seems to me, that setting `g:lsp_ale_auto_enable_linter` to `false` is ignored. 

When I remove the plugin, I'm able again to lint with `vls` again.

Maybe I missed something?

Thanks
Marc

**Comment by idbrii** [comment]
That is how it used to work. Try running this in a vue file:

    :let active_linters = get(b:, 'ale_linters', get(g:ale_linters, &filetype, []))
    :echomsg index(active_linters, 'vim-lsp')
    :echomsg active_linters g:lsp_ale_auto_enable_linter

That's roughly how s:is_active_linter checks whether we should lint:

https://github.com/rhysd/vim-lsp-ale/blob/db0f9a8a33c0480988dc420cd2fba8a07743e4a4/autoload/lsp/ale.vim#L74-L80

However, maybe ale is doing something new because it detects vim-lsp-ale's presence? I don't understand how vls is both enabled and ignored. What does `:ALEInfo` output if you remove vim-lsp-ale?

**Comment by dubst3pp4** [comment]
I've just tried the same setup (vim-lsp + ale + vim-lsp-ale and vls) again and it seems to work without problems now.

I removed the redundant `vls` setting from `b:ale_linters` so I just  have:

```vim
let b:ale_linters = ['eslint']
let b:ale_fixers = ['eslint', 'vls']
```

in my `after/ftplugin/vue.vim.

Luckily, ALE now reports errors by `vls`, as it seems that `vim-lsp-ale` automatically uses the LSP service started by `vim-lsp`:
![Bildschirmfoto von 2022-06-14 13-50-30](https://user-images.githubusercontent.com/2373731/173570582-a58bdff5-8b16-4973-bfc8-a9eed4e56abd.png)

I will close this issue.
